htsohm/htsohm_hpc.py

```diff
@@ -9,37 +9,41 @@
 job_queue = sjs.get_job_queue()
 
 from htsohm.utilities import write_config_file, read_config_file
-from htsohm.htsohm import seed_generation, next_generation, hpc_job_run_all_simulations
+from htsohm.htsohm import seed_generation, queue_all_materials, queue_create_next_gen
+from htsohm.htsohm import update_strength_array
 from htsohm.runDB_declarative import Material, session
 
 def start_run(
         children_per_generation,    # number of materials per generation
         number_of_atomtypes,        # number of atom-types per material
         strength_0,                 # intial strength parameter
         number_of_bins,             # number of bins for analysis
-        max_generations=20):        # maximum number of generations
+        max_generations=20,         # maximum number of generations
+        dummy_test_trials=3,        # number of re-simulations for dummy-test
+        acceptance_value=-0.001):    # desired degree of `convergence`
 
     run_id = write_config_file(children_per_generation, number_of_atomtypes, strength_0,
-        number_of_bins, max_generations)['run-id']
+        number_of_bins, max_generations, dummy_test_trials, acceptance_value)['run-id']
 
     return run_id
 
 def manage_run(run_id, generation):
     config = read_config_file(run_id)
-
+#    acceptance_value = config['acceptance-value']
+#    convergence = acceptance_value + 1
+    
     # prepare, mutate, and queue up the next generation
     if generation > config['max-number-of-generations']:
-        print("max generations exceeded; we're done here!")
+        print("Max generations exceeded; terminating run.")
         return -1 # -1 means we're done
     elif generation == 0:
         # SEED GENERATION
         seed_generation(run_id,
-                        config['children-per-generation'], config['number-of-atom-types'],
-                        queue=job_queue)
+                        config['children-per-generation'], config['number-of-atom-types'])
+        queue_all_materials(run_id, generation, queue)
     elif generation >= 1:
         # FIRST GENERATION, AND ON...
-        next_generation(run_id,
-                        config['children-per-generation'], generation,
-                        queue=job_queue)
-
+        update_strength_array(run_id, generation)
+        queue_create_next_gen(run_id, generation, queue)
+        queue_all_materials(run_id, generation, queue)
```

Comment:
Careful here. This will queue up the create_next_gen jobs and the queue_all_materials job all at once. So queue_all_materials will be run before any of the create_next_gen jobs have been completed (or even likely started). So I think that will mean that no materials will be queued.